### DISCUSSION TITLE
LDM notes for April 15th 2020

### DISCUSSION THREAD

**Discussion by YairHalberstadt** [dotnet/csharplang#3413] [Language Ideas]
https://github.com/dotnet/csharplang/blob/master/meetings/2020/LDM-2020-04-15.md

1. Non-void and non-private partial methods
2. Top-level programs

**Comment by YairHalberstadt** [comment]
> Another question is whether we would allow other members to be partial. We would need to work out the syntax in each case: E.g. how do you distinguish a partial property definition from a declaration that implements it as an auto-property?

Presumably using the partial keyword on the property just like methods? What other syntax could there possibly be? What else could such a syntax possibly mean?

> On the other hand we are not ready to allow partial on other kinds of members. If future scenario bear out a strong need, we will do the design work to hash it out, but we think methods are able to address the vast majority of what's needed.

Here is a list of requests with the aim of simplifying usage of NotifyPropertyChanged:
#2657
#821 (see https://github.com/dotnet/csharplang/issues/821#issuecomment-322198258)
https://github.com/dotnet/csharplang/issues/3329#issuecomment-610276756
https://github.com/dotnet/csharplang/issues/2976
https://github.com/dotnet/csharplang/issues/1419

Writing NotifyPropertyChanged for properties which could otherwise be auto properties is clearly a significant burden for users, and partial properties + source generators are the perfect solution.

**Comment by tannergooding** [comment]
I imagine with properties you would put the partial modifier on the actual `get`/`set` similar to what you can do for `readonly`:
```csharp
public int X
{
    partial get;
    partial set;
}
```

Then if it is redeclared without a `partial` modifier it must be `auto`

**Reply by alrz** [reply]
What's the problem with 
```cs
partial class X { public partial Prop {get;set;} }
partial class X { public Prop {get;set;} }
```
?

**Reply by maxkoshevoi** [reply]
Hm, I guess, problem is that currently `partial` modifier also needs to be included on the implementation part. So it's not obvious, that you want to implement property as auto-property
